### PR TITLE
Added support for Spark COLLECT_LIST and CONCAT_WS function and STANDARD_HASH and BITANDOR and DB2 TIMESTAMP_TRUNC functions

### PR DESCRIPTION
**Pull request by Kowsikraja**

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java [hunk 1]
```diff
@@ -11590,6 +11590,33 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
...
 
+  @Test public void testStandardHash() {
+    final RelBuilder builder = relBuilder();
+    final RexNode stdHash = builder.call(SqlLibraryOperators.STANDARD_HASH,
+        builder.scan("EMP").field("ENAME"));
+    final RelNode root = builder
+        .scan("EMP")
+        .project(stdHash)
+        .build();
+    final String expectedBQSql = "SELECT STANDARD_HASH(\"ENAME\")"
```
Comment (nosheenSheikh02) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java:11601]: change this to `expectedOracleSql`

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java [hunk 2]
```diff
@@ -11600,6 +11627,22 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
...
+    final RelBuilder builder = relBuilder();
+    final RexNode bitPart = builder.call(SqlLibraryOperators.BITANDNOT,
+        builder.literal(3), builder.literal(2));
+
+    final RelNode root = builder
+        .scan("EMP")
+        .project(builder.alias(bitPart, "Result"))
+        .build();
+
+    final String expectedQuery = "SELECT BITANDNOT(3, 2) AS Result\nFROM scott.EMP AS EMP";
```
Comment (nosheenSheikh02) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java:11640]: change this to `expectedDB2Query`

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java [hunk 3]
```diff
@@ -12522,6 +12565,20 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
...
+  @Test public void testForTruncTimestampFunction() {
+    final RelBuilder builder = relBuilder();
+    final RexNode truncTimestampNode = builder.call(SqlLibraryOperators.TRUNC_TIMESTAMP,
+        builder.call(CURRENT_TIMESTAMP),
+        builder.literal("Year"));
+    final RelNode root = builder
+        .scan("EMP")
+        .project(truncTimestampNode)
+        .build();
+    final String expectedBQSql = "SELECT TRUNC_TIMESTAMP(CURRENT_TIMESTAMP, 'Year') AS $f0\nFROM"
```
Comment (nosheenSheikh02) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java:12577]: change this to `expectedDB2Sql`

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java [hunk 4]
```diff
@@ -12522,6 +12565,20 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
...
+        + " scott.EMP AS EMP";
+
+    assertThat(toSql(root, DatabaseProduct.DB2.getDialect()), isLinux(expectedBQSql));
+  }
```
Comment (nosheenSheikh02) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java:12581]: Add test case for `Concat_Ws` , `Collect_list` spark functions
and  `BitNot` for DB2

**Review by nosheenSheikh02 — CHANGES_REQUESTED** [review]

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java (hunk 4 above)
Comment (Kowsikraja) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java:12581]: @nosheenSheikh02 I had added the test cases.Please have a look.

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java (hunk 1 above)
Comment (Kowsikraja) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java]: @nosheenSheikh02 Chaged the name.Please have a look.

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java (hunk 2 above)
Comment (Kowsikraja) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java]: @nosheenSheikh02 Chaged the name.Please have a look.

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java (hunk 3 above)
Comment (Kowsikraja) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java]: @nosheenSheikh02 Chaged the name.Please have a look.

**Review by nosheenSheikh02 — APPROVED** [review]

**Review by anuragsaxena2k7 — APPROVED** [review]
approve

**Pull request merged**